nerfstudio/engine/trainer.py

```diff
@@ -456,10 +459,16 @@ def train_iteration(self, step: int) -> TRAIN_INTERATION_OUTPUT:
 
         device_type: str = self.device.split(":")[0] if "cuda" in self.device else "cpu"
 
-        with torch.autocast(device_type=device_type, enabled=self.mixed_precision):
-            _, loss_dict, metrics_dict = self.pipeline.get_train_loss_dict(step=step)
-            loss = functools.reduce(torch.add, loss_dict.values())
-        self.grad_scaler.scale(loss).backward()  # type: ignore
+        internal_step = 0
+        while True:
```

Comment:
This would be clearer with a for loop, since internal_step is unnecessary except for counting iterations